paella-matterhorn/javascript/05_initdelegate.js

```diff
@@ -58,22 +55,25 @@ function initPaellaMatterhorn(episodeId, onSuccess, onError) {
 					       paella.matterhorn.resourceId = "";
 					   }
 					   // end #DCE logging helper
-						
-						if (serie != undefined) {
-							asyncLoader.addCallback(new paella.JSONCallback({url:'/series/'+serie+'.json'}), "serie");
-							asyncLoader.addCallback(new paella.JSONCallback({url:'/series/'+serie+'/acl.json'}), "acl");
-						
-							asyncLoader.load(function() {
-									//Check for series
-									paella.matterhorn.serie = asyncLoader.getCallback("serie").data;
-									//Check for acl
-									paella.matterhorn.acl = asyncLoader.getCallback("acl").data;
-									if (onSuccess) onSuccess();
-								},
-								function() {
-									if (onError) onError();
-								}
-							);
+					   // start #DCE get series from serch endpoint (instead of series endpoint)
+					   if (serie != undefined) {
+					       base.ajax.get({
+					           url: '/search/series.json',
+					           params: {'id': serie }
+					           },
+					           function (data, contentType, code) {
+					           var jsonData = data;
+					           if (typeof (jsonData) == "string") jsonData = JSON.parse(jsonData);
+					           // #DCE verify that results returned at least one series
+					           var totalItems = parseInt(data[ 'search-results'].total);
+					           if (totalItems === 0) {
+					               showLoadErrorMessage(paella.dictionary.translate("No series found for series id") + ": \"" + serie + "\"");
+					               return;
+					           }
+					           searchSeriesToSeriesSeries(data[ 'search-results'].result);
+					           // end #DCE get series from search endpoint
+					           if (onSuccess) onSuccess();
+					       });
 						}
 						else {
 							if (onSuccess) onSuccess();						
```

Comment:
Since `onError` is never called anymore, the `onSuccess` call should probably sit outside of the if…else block. If it is the case that onError never needs to be called.